### PR TITLE
precheck(dm): refine error message & instruction

### LINKED ISSUE #7621: DM: refine error message in precheck
For example, the solution of `table compatibility` is placed in each table's `Error.Instruction`. It should've been in the `Result`'s instruction.

### PR DESCRIPTION
**Pull request by buchuitoudegou**
### What problem does this PR solve?

Issue Number: close #7621 

### What is changed and how it works?
- refine error message and instruction
- merge instruction for each table in `table compatibility check`

#### Tests 

 - Unit test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by buchuitoudegou** [comment]
/area dm
/run-dm-integration-test

the message and instruction need to be confirmed by Frank

File: dm/pkg/checker/binlog.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func (pc *MySQLBinlogFormatChecker) Check(ctx context.Context) *Result {
 	}
 	if strings.ToUpper(value) != "ROW" {
 		result.Errors = append(result.Errors, NewError("binlog_format is %s, and should be ROW", value))
-		result.Instruction = "please execute 'set global binlog_format=ROW;'"
+		result.Instruction = "MySQL as source: please execute 'set global binlog_format=ROW; AWS Aurora (MySQL)/RDS MySQL as source : please refer to the document to create a new DB parameter group and set the binlog_format=row: https://docs.aws.amazon.com/zh_cn/AmazonRDS/latest/AuroraUserGuide/USER_WorkingWithDBInstanceParamGroups.html. Then modify the instance to use the new DB parameter group and restart the instance to take effect."
```
Comment (okJiang) [inline dm/pkg/checker/binlog.go:97]:
```suggestion
		result.Instruction = "MySQL as source: please execute 'set global binlog_format=ROW;'; AWS Aurora (MySQL)/RDS MySQL as source: please refer to the document to create a new DB parameter group and set the binlog_format=row: https://docs.aws.amazon.com/zh_cn/AmazonRDS/latest/AuroraUserGuide/USER_WorkingWithDBInstanceParamGroups.html. Then modify the instance to use the new DB parameter group and restart the instance to take effect."
```

File: dm/pkg/checker/binlog.go [hunk 2]
```diff
@@ -172,7 +172,7 @@ func (pc *MySQLBinlogRowImageChecker) Check(ctx context.Context) *Result {
 	}
 	if strings.ToUpper(value) != "FULL" {
 		result.Errors = append(result.Errors, NewError("binlog_row_image is %s, and should be FULL", value))
-		result.Instruction = "please execute 'set global binlog_row_image = FULL;'"
+		result.Instruction = "MySQL as source: please execute 'set global binlog_row_image = FULL;' AWS Aurora (MySQL)/RDS MySQL as source : please refer to the document to create a new DB parameter group and set the binlog_row_image = FULL: https://docs.aws.amazon.com/zh_cn/AmazonRDS/latest/AuroraUserGuide/USER_WorkingWithDBInstanceParamGroups.html Then modify the instance to use the new DB parameter group and restart the instance to take effect."
```
Comment (okJiang) [inline dm/pkg/checker/binlog.go:175]:
```suggestion
		result.Instruction = "MySQL as source: please execute 'set global binlog_row_image = FULL;'; AWS Aurora (MySQL)/RDS MySQL as source: please refer to the document to create a new DB parameter group and set the binlog_row_image = FULL: https://docs.aws.amazon.com/zh_cn/AmazonRDS/latest/AuroraUserGuide/USER_WorkingWithDBInstanceParamGroups.html Then modify the instance to use the new DB parameter group and restart the instance to take effect."
```

File: dm/pkg/checker/mysql_server.go [hunk 3]
```diff
@@ -72,10 +74,14 @@ func (pc *MySQLVersionChecker) Check(ctx context.Context) *Result {
 func (pc *MySQLVersionChecker) checkVersion(value string, result *Result) *Error {
 	needVersion := SupportedVersion["mysql"]
 	if utils.IsMariaDB(value) {
-		return NewWarn("Migrating from MariaDB is experimentally supported. If you must use DM to migrate data from MariaDB, we suggest make your MariaDB >= 10.1.2")
+		err := NewWarn("Migrating from MariaDB is still experimental. It is recommended that you upgrade MariaDB to 10.1.2 or a later version.")
+		err.Instruction = "It is recommended that you upgrade MariaDB to10.1.2 or a later version."
```
Comment (okJiang) [inline dm/pkg/checker/mysql_server.go:78]: repeated?

**Review by okJiang — COMMENTED** [review]
rest lgtm

File: dm/pkg/checker/binlog.go (hunk 2 above)
Comment (buchuitoudegou) [reply dm/pkg/checker/binlog.go:175]: The message is given by docs team. I will argue with them later

**Review by D3Hunter — APPROVED** [review]

**Review by okJiang — APPROVED** [review]
Function and text are LGTM

**Comment by buchuitoudegou** [comment]
cdc unstable test? https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftiflow%2Fghpr_verify/detail/ghpr_verify/7/pipeline/149

/run-verify

**Pull request merged**